chat-plugins/blackjack.js

```diff
@@ -0,0 +1,501 @@
+			this.display(Chat.html`<br /><strong>${this.players[this.curUser].name}</strong> stands with ${cards} (${this.players[this.curUser].points}) (Auto-stand: took too long to move)`);
+			this.next();
+		}, this.turnTimeoutMinutes * 60 * 1000);
+	}
+}
+
+class BlackjackPlayer extends Rooms.RoomGamePlayer {
+	constructor(user, game) {
+		super(user, game);
+
+		this.cards = [];
+		this.points = 0;
+		this.status = 'playing';
+		this.selfUhtml = '';
+	}
+}
+
+exports.commands = {
+	bj: 'blackjack',
+	blackjack: {
+		new: 'create',
+		create: function (target, room, user) {
+			if (!this.can('gamemanagement', null, room)) return;
+			if (room.game) return this.errorReply("There is already a game running in this room.");
+			if (room.blackjackDisabled) return this.errorReply("Blackjack is currently disabled in this room.");
+
+			room.game = new Blackjack(room, user, target);
+		},
+		start: function (target, room, user) {
+			if (!this.can('gamemanagement', null, room)) return;
+			if (!room.game || room.game.title !== 'Blackjack') return this.errorReply("There is no game of blackjack currently ongoing in this room.");
+
+			room.game.start(user);
+		},
+		forceend: 'end',
+		end: function (target, room, user, connection, cmd) {
+			if (!this.can('gamemanagement', null, room)) return;
+			if (!room.game || room.game.title !== 'Blackjack') return this.errorReply("There is no game of blackjack currently ongoing in this room.");
+
+			room.game.end(user, cmd);
+		},
+		hit: function (target, room, user) {
+			if (!room.game || room.game.title !== 'Blackjack') return this.errorReply("There is no game of blackjack currently ongoing in this room.");
+
+			room.game.hit(user);
+		},
+		stand: function (target, room, user) {
+			if (!room.game || room.game.title !== 'Blackjack') return this.errorReply("There is no game of blackjack currently ongoing in this room.");
+
+			room.game.stand(user);
+		},
+		j: 'join',
+		join: function (target, room, user) {
+			return this.parse('/joingame');
+		},
+		l: 'leave',
+		leave: function (target, room, user) {
+			return this.parse('/leavegame');
+		},
+		unspectate: 'spectate',
+		spectate: function (target, room, user, connection, cmd) {
+			if (!room.game || room.game.title !== 'Blackjack') return this.errorReply("There is no game of blackjack currently ongoing in this room.");
+
+			if (cmd === 'spectate') {
+				room.game.spectate(user);
+			} else if (cmd === 'unspectate') {
+				room.game.unspectate(user);
+			}
+		},
+		disable: function (target, room, user) {
+			if (!this.can('declare', null, room)) return;
+			if (room.blackjackDisabled) return this.errorReply("Blackjack is already disabled in this room.");
+			room.blackjackDisabled = true;
+			this.privateModCommand(`(${user.name} disabled games of blackjack in this room.)`);
+
+			if (room.chatRoomData) {
+				room.chatRoomData.blackjackDisabled = room.blackjackDisabled;
+				Rooms.global.writeChatRoomData();
```

Comment:
`GlobalRoom#writeChatRoomData` saves a ton of data unrelated to this plugin. It'd be better to write to a file in `config/chat-plugins/` instead of coupling the plugin to chatrooms like this